docs/how-to/rocks/add-internal-user-in-rocks.rst

```diff
@@ -0,0 +1,45 @@
+.. _add_internal_user_in_rocks:
+
+How to add an internal user to a rock
+*************************************
+
+Rockcraft allows you to use ``run-user`` to specify the user you want to the
+rock's service(s) with. Without using that field in your rockcraft.yaml, the
+service(s) will be run by the root user. The downside, however, of using
+``run-user`` is that only supports one user called ``_daemon_`` with a UID/GID
+of 84792. For some daemon services that will be enough, but some other services
+require that the running user should have a specific username and/or UID.
+
+In order to add new internal user or group to a rock, three packages (or their
+respective slices) are needed:
+- ``base-files``: to create the base directories such as ``/etc`` and ``/bin``
+- ``base-passwd``: to produce basic ``/etc/passwd`` and ``/etc/group`` files,
+- ``passwd``: to provide ``useradd`` and ``groupadd`` commands.
```

Comment:
I don't understand why this needs to be staged at all. The `useradd` and `groupadd` commands being used in `override-build` come from the build system, don't they?

Reply:
this is needed for /root and /home directories. Without /root, the default root user (that has /root as its home) won't be able to run any commands (an error will show because /root does not exist). Similarly, /home is needed to become the base dir of the home dir of the created user(s).

Reply:
right but I think @lengau is asking specifically about `passwd` - the build system already has `useradd` and `groupadd`

Reply:
removed passwd_bins slice.